test/src/test/java/org/corfudb/runtime/object/transactions/SnapshotTransactionContextTest.java

```diff
@@ -87,6 +90,22 @@ public void testSnapshotTxNestedExplicitTx() {
         }));
         t(0, this::TXEnd);
         t(0, this::TXEnd);
+    }
 
+
+    /**
+     * Ensure that TransactionAbortedException(AbortCause.UNSUPPORTED_OPERATION) is thrown when
+     * doing a mutation inside of a snapshot read.
+     */
+    @Test
+    public void testUnsupportedOperationException() {
+        try {
+            SnapshotTXBegin();
+            put("k" , "v1");
```

Comment:
I thought the problem was the endless retrying in an implicit transaction while being in a Snapshot Transaction.
This is not what is tested here.

You can test something like this:
```
SnapshotTXBegin();
putIfAbsent("k", "v1);
TXEnd();
```